Assets/Src/Scripts/SeedCalc/RequestReviewUtils.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright 2021-2022 The SeedV Lab.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using UnityEngine;
+
+#if UNITY_IOS
+using UnityEngine.iOS;
+#endif
+
+namespace SeedCalc {
+  public static class RequestReviewUtils {
+    private const string _userPrefKeyCounter = "process-completed-counter";
+    private const string _userPrefKeyLastVersion = "last-version-promoted-for-review";
+    private const int _threshold = 50;
+
+    // Increases the counter stored in the user preferences. If the counter exceeds the threshold
+    // for the current version, invokes the API call to request an app store review.
+    //
+    // See https://developer.apple.com/documentation/storekit/requesting_app_store_reviews for a
+    // reference code.
+    public static void CountAndRequestReviewIfNecessary() {
+
+      // For now, only iOS build has this feature enabled.
+      #if UNITY_IOS
+
+      int counter = PlayerPrefs.GetInt(_userPrefKeyCounter, 0);
+      PlayerPrefs.SetInt(_userPrefKeyCounter, ++counter);
```

Comment:
The count is still increasing when `Application.version == lastVersion`. If the counter is larger than 0, like `_threshold - 1`, then the user upgrades to a new version, there will be a problem that the request will prompt immediately. Is it right?

Reply:
You are right. The current logic only checks one preference parameter (the counter) until the threshold is met. This looks simpler than checking both the counter and the version number every time, or some other mixed logic.

I think it is not a big deal whether the counter is accurate or not. It's just a hit number and whether to show the dialog is actually controlled by iOS SDK. We cannot even make sure or know it is indeed shown every time the threshold is reached.

Thus I'd suggest keeping this simple logic. Sounds good?